p5/core/vertex.py

```diff
@@ -210,15 +241,30 @@ def end_contour():
 
     For more info, see :any:`begin_contour`.
     """
-    global is_contour, curr_contour_vertices, curr_contour_vertices_types
-    is_contour = False
-    # Close contour
-    curr_contour_vertices.append(curr_contour_vertices[0])
-    curr_contour_vertices_types.append(curr_contour_vertices_types[0])
-    # Save contour
-    contour_vertices.append(curr_contour_vertices)
-    contour_vertices_types.append(curr_contour_vertices_types)
-    curr_contour_vertices, curr_contour_vertices_types = [], []
+    global is_contour, curr_contour_vertices, curr_contour_vertices_types, is_first_contour
+
+    if builtins.current_renderer == 'vispy':
+        # Close contour
+        curr_contour_vertices.append(curr_contour_vertices[0])
+        curr_contour_vertices_types.append(curr_contour_vertices_types[0])
+        # Save contour
+        contour_vertices.append(curr_contour_vertices)
+        contour_vertices_types.append(curr_contour_vertices_types)
+        curr_contour_vertices, curr_contour_vertices_types = [], []
+    elif builtins.current_renderer == 'skia':
+
+        vert_data = copy.deepcopy(contour_vertices[0])
+        vert_data[-1]["is_vert"] = contour_vertices[0][-1].get('is_vert', None)
+        vert_data[-1]["move_to"] = False
+
+        contour_vertices.append(vert_data)
+
+        if is_first_contour:
+            vertices.append(vertices[0])
```

Comment:
What was the reaons to append `vertices[0]` for the first contour, but not subsequent contours?

Reply:
Close the shape before starting the contour, else we will get a line connecting the contour and the shape.

Reply:
Got it! Could you document this in an inline comment?